controller/api/v1alpha1/deploymentsettings_types.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	console "github.com/pluralsh/console-client-go"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// DeploymentSettingsSpec defines the desired state of DeploymentSettings
+type DeploymentSettingsSpec struct {
+	// AgentHelmValues custom helm values to apply to all agents (useful for things like adding customary annotations/labels)
+	// +kubebuilder:validation:Optional
+	AgentHelmValues *runtime.RawExtension `json:"agentHelmValues,omitempty"`
+
+	// Stacks global configuration for stack execution
+	Stacks *StackSettings `json:"stacks,omitempty"`
+
+	// Bindings
+	// +kubebuilder:validation:Optional
+	Bindings *DeploymentSettingsBindings `json:"bindings,omitempty"`
+
+	// PrometheusConnection connection details for a prometheus instance to use
+	PrometheusConnection *HTTPConnection `json:"prometheusConnection,omitempty"`
+
+	// connection details for a loki instance to use
+	LokiConnection *HTTPConnection `json:"lokiConnection,omitempty"`
+}
+
+type HTTPConnection struct {
+	Host string `json:"host"`
+	// user to connect w/ for basic auth
+	User *string `json:"user,omitempty"`
+	// password to connect w/ for basic auth
+	Password *string `json:"password,omitempty"`
+}
+
+func (r *HTTPConnection) Attributes() *console.HTTPConnectionAttributes {
+	return &console.HTTPConnectionAttributes{
+		Host:     r.Host,
+		User:     r.User,
+		Password: r.Password,
+	}
+}
+
+type DeploymentSettingsBindings struct {
+	// Read bindings.
+	// +kubebuilder:validation:Optional
+	Read []Binding `json:"read,omitempty"`
+
+	// Write bindings.
+	// +kubebuilder:validation:Optional
+	Write []Binding `json:"write,omitempty"`
+
+	// Create bindings.
+	// +kubebuilder:validation:Optional
+	Create []Binding `json:"create,omitempty"`
+
+	// Git bindings.
+	// +kubebuilder:validation:Optional
+	Git []Binding `json:"git,omitempty"`
+}
+
+type StackSettings struct {
```

Comment:
We should also add `ConnectionRef` here mapping to `connectionId` in the stack settings

Reply:
done